### DISCUSSION TITLE
Error: n is not a function

### DISCUSSION THREAD

**Discussion by EmilMalanczak** [vercel/next.js#17467] [Help]
# Bug report

## Describe the bug

For a couple of days my Sentry.io is spamming about confusing error `n is not a function`.  I've found exact place where it's getting executed but nothing comes to my mind what would be a solution for that issue

place: 

```
function vo(e, t, n) {
    switch ("function" === typeof Nu && Nu(t), t.tag) {
        case 0:
        case 11:
        case 14:
        case 15:
            if (null !== (e = t.updateQueue) && null !== (e = e.lastEffect)) {
                var r = e.next;
                Hl(97 < n ? 97 : n, (function () {
                    var e = r;
                    do {
                        var n = e.destroy;
                        if (void 0 !== n) {
                            var l = t;
                            try {
                                n()
                            } catch (a) {
                                Su(l, a)
                            }
                        }
                        e = e.next
                    } while (e !== r)
                }))
            }
            break;
        case 1:
            po(t), "function" === typeof (n = t.stateNode).componentWillUnmount && function (e, t) {
                try {
                    t.props = e.memoizedProps, t.state = e.memoizedState, t.componentWillUnmount()
                } catch (n) {
                    Su(e, n)
                }
            }(t, n);
            break;
        case 5:
            po(t);
            break;
        case 4:
            ko(e, t, n)
    }
}
```

## To Reproduce

I've tried to reproduce that error but no results. Everything works as expected, every page is rendering correctly. 

## Expected behavior

No error.

## Screenshots

![image](https://user-images.githubusercontent.com/53233583/94664755-91c40b80-030b-11eb-85a2-930a95ba2a4e.png)

## System information

- OS & Browser: No rule, noticed that error on all OS and browsers (including IE 11) + mobile devices
- Version of Next.js: 9.5.3

## Additional context

After analysis of sentry's notifications I noticed that a lot of error are repetitive on same IP.

**Comment by jamesmosier** [comment]
Hi @EmilMalanczak. Do you happen to have any code from your project you could share? I'm not sure if it'll be possible to help out with this issue otherwise!

**Comment by EmilMalanczak** [comment]
Hi @jamesmosier! 
The problem is that I don't really know what part of code is responsible for it. Sharing whole project is impossible, do you have any idea which file or function might be a problem?  I might send some screenshots from files like _app.js

**Reply by jamesmosier** [reply]
Unfortunately I have no clue without seeing some code. But looks like Sentry provides a nice trace of what page/actions lead up to the error.

**Reply by binar123** [reply]
but the biggest problem is we cannot reproduce this error on any environment so any tip when this function is called would be good for us

**Comment by EmilMalanczak** [comment]
### Found  a solution - the problem is somehow connected with this component.

![image](https://user-images.githubusercontent.com/53233583/95326856-8476c600-08a3-11eb-90ea-ca402991559d.png)

It uses 2 custom hooks: 
- useAnimation just returns styles by material makeStyles
- useOnScreen which is based on IntersectionObserver

This is that 2nd hook code:

![image](https://user-images.githubusercontent.com/53233583/95327344-2dbdbc00-08a4-11eb-808c-fa175618a18a.png)

Error was showing only on 1 single page. There was no problem on any others
Any idea what is a cause of that error?
